### PR TITLE
Fix:#309 4 Cancel toast on finish

### PR DESCRIPTION
**Pull request by vishwesh3**
PR addressing #309 4

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@vishwesh3, it looks like you've already signed the Dropbox CLA. Thanks!

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 1]
```diff
@@ -1633,6 +1637,28 @@ public void clearSearch() {
         inSearch = false;
     }
 
+    /**
+     * dismiss toast
+     */
+    private void dismissToast() {
+        try {
+            toast.cancel();
+        } catch (NullPointerException e) {
```
Comment (Sam1301) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: hear i think we could handle this error gracefully by `if (toast != null)` check and not send the traceback to crashylytics using `Zlog.logException(e)` :)

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 2]
```diff
@@ -1633,6 +1637,28 @@ public void clearSearch() {
...
+            ZLog.logException(e);
+        }
+    }
+
+    /**
+     * If previous toast is showing cancel it and show new one
+     * @param string message in the toast
```
Comment (Sam1301) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: second param "duration" description missing from javadoc comment :)

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java [hunk 3]
```diff
@@ -1624,6 +1627,7 @@ public void run() {
         }
     }
 
+
```
Comment (vishwesh3) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: Done :)

File: app/src/main/java/com/zulip/android/activities/ZulipActivity.java (hunk 1 above)
Comment (vishwesh3) [inline app/src/main/java/com/zulip/android/activities/ZulipActivity.java]: Done :)

**Comment by Sam1301** [comment]
@vishwesh3 Nice, the code looks good to me :+1:

**Comment by timabbott** [comment]
Merged, thanks @vishwesh3 and @Sam1301!  Also edited the commit message to avoid closing #309 since it's not done yet.

**Pull request closed without merging**